test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -79,167 +82,313 @@ func (s *snapshottableTestSuite) SkipRedundantSuite(driver TestDriver, pattern t
+			// typically this snapshot has already been deleted
+			err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Delete(context.TODO(), l.vs.GetName(), metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				framework.Failf("Error deleting snapshot %q. Error: %v", l.pvc.Name, err)
+			}
+		})
+		err = WaitForSnapshotReady(l.dc, l.vs.GetNamespace(), l.vs.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("checking the snapshot")
+		// Get new copy of the snapshot
+		l.vs, err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Get(context.TODO(), l.vs.GetName(), metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		// Get the bound snapshotContent
+		snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+		snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+		l.vscontent, err = l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		snapshotContentSpec := l.vscontent.Object["spec"].(map[string]interface{})
+		volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+
+		// Check SnapshotContent properties
+		ginkgo.By("checking the SnapshotContent")
+		framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], l.vsc.GetName())
+		framework.ExpectEqual(volumeSnapshotRef["name"], l.vs.GetName())
+		framework.ExpectEqual(volumeSnapshotRef["namespace"], l.vs.GetNamespace())
+	}
+
+	TestSnapshotDeleted := func(sct SnapshotClassTest) {
+		var err error
+		snapshotStatus := l.vs.Object["status"].(map[string]interface{})
+		snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+
+		ginkgo.By("deleting the snapshot")
+		err = l.dc.Resource(SnapshotGVR).Namespace(l.vs.GetNamespace()).Delete(context.TODO(), l.vs.GetName(), metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Error deleting snapshot %s in namespace %s. Error: %v", l.vs.GetName(), l.vs.GetNamespace(), err)
 		}
-	})
 
-	// This intentionally comes after checking the preconditions because it
-	// registers its own BeforeEach which creates the namespace. Beware that it
-	// also registers an AfterEach which renders f unusable. Any code using
-	// f must run inside an It or Context callback.
-	f := framework.NewDefaultFramework("snapshotting")
+		if sct.DeletionPolicy == "Delete" {
+			ginkgo.By("checking the SnapshotContent has been deleted")
+			err = WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, framework.Poll, framework.SnapshotDeleteTimeout)
+			framework.ExpectNoError(err)
+		} else if sct.DeletionPolicy == "Retain" {
+			ginkgo.By("checking the SnapshotContent has not been deleted")
+			_, err = l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
```

Comment:
Do we need to add a wait for some amount of time before checking the content is not deleted?  Otherwise, the test may check too early and the content gets deleted sometime later.

Reply:
Updated to wait for 10 second, retrying every second.